hudi-common/src/main/java/org/apache/hudi/common/table/read/BufferedRecord.java

```diff
@@ -41,12 +42,12 @@
  */
 public class BufferedRecord<T> implements Serializable {
   private final String recordKey;
-  private final Comparable orderingValue;
+  private final Lazy<Comparable> orderingValue;
```

Comment:
When deriving the BufferedRecord from the record with `emitDeletes` we can run into issues trying to extract the ordering value since the record is null. This allows us to only perform that evaluation if required.

Reply:
We already cache the `orderingVal` in `HoodieRecord`, the `Lazy` wrapper will increase the bytes to serialize for the spillableMap too.